rust/src/wallet.rs

```diff
@@ -301,7 +301,39 @@ pub async fn open_cfd(taker_amount: u64, maker_amount: u64) -> Result<()> {
         .map_err(|e| anyhow!(e))?;
     tracing::info!(?custom_output_details, "Added custom output");
 
-    // TODO: Persist custom output for collab close
+    Ok(())
+}
+
+pub async fn settle_cfd(taker_amount: u64, maker_amount: u64) -> Result<()> {
+    tracing::info!("Settling CFD with taker amount {taker_amount} and maker amount {maker_amount}");
+
+    let (peer_manager, channel_manager) = {
+        let lightning = &get_wallet()?.lightning;
+        (
+            lightning.peer_manager.clone(),
+            lightning.channel_manager.clone(),
+        )
+    };
+
+    let custom_output_id = *channel_manager
+        .custom_outputs()
+        .first()
+        .context("No custom outputs in channel")?;
+
+    let channels = channel_manager.list_channels();
+    let channel_details = channels.first().context("No channels found")?;
+    let maker_pk = channel_details.counterparty.node_id;
+
+    let taker_amount_msats = taker_amount * 1000;
+    let maker_amount_msats = maker_amount * 1000;
+
+    let maker_connection_str = format!("{maker_pk}@{MAKER_IP_PORT}");
+    let peer_info = parse_peer_info(maker_connection_str)?;
+    connect_peer_if_necessary(&peer_info, peer_manager.clone()).await?;
```

Comment:
This shouldn't be necessary. Afaik the node will automatically reconnect on restart if there is an open channel. If there is no open channel to the maker, then we shouldn't be in this function anyway :)

Reply:
True, we actually found out that it does on my machine and it was a diffferent issue, we could remove this.

Reply:
Done with https://github.com/itchysats/10101/pull/207/commits/b961223edf3a7fba28e1368303f11b1be8319a04.